sale_order_type/models/sale_order.py

```diff
@@ -0,0 +1,38 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def _get_order_type(self):
+        return self.env['sale.order.type'].search([])[:1]
+
+    type_id = fields.Many2one(
+        comodel_name='sale.order.type', string='Type', default=_get_order_type)
+
+    @api.multi
+    def on_change_type_id(self, type_id):
```

Comment:
Because it was done before new api, and we did not take that into account, it was migrated quickly and we miss it. Thanks for the pointing. :wink:

I'll take a look